src/backend/distributed/utils/attribute.c

```diff
@@ -619,3 +604,126 @@ MultiTenantMonitorshmemSize(void)
 
 	return size;
 }
+
+
+/*
+ * ExtractTopComment extracts the top-level multi-line comment from a given input string.
+ */
+static char *
+ExtractTopComment(const char *inputString)
+{
+	int commentStartCharsLength = 2;
+	int inputStringLen = strlen(inputString);
+	if (inputStringLen < commentStartCharsLength)
+	{
+		return NULL;
+	}
+
+	int commentEndCharsIndex = 0;
+
+	/* If query doesn't start with a comment, return NULL */
+	if (inputString[commentEndCharsIndex] != '/' ||
+		inputString[commentEndCharsIndex + 1] != '*')
+	{
+		return NULL;
+	}
+
+	/* Skip the comment start characters */
+	commentEndCharsIndex += commentStartCharsLength;
+	while (commentEndCharsIndex < inputStringLen &&
```

Comment:
nit: The way this the comment was attached to the lines below made me think on a quick look that the loop was also skipping the comment characters. Adding another comment and/or a newline would have prevented that.

```suggestion
	/* Skip the comment start characters */
	commentEndCharsIndex += commentStartCharsLength;
	
	/* Find the first comment end character */
	while (commentEndCharsIndex < inputStringLen &&
```